test/suite_implementation.js

```diff
@@ -65,29 +67,26 @@ module.exports = function(style, options, callback) {
 };
 
 function applyOperations(map, operations, callback) {
-    var operation = operations && operations[0];
-
-    if (!operations || operations.length === 0) {
-        map.once('render', function onRender() {
-            if (map.loaded()) {
-                callback();
-            } else {
-                map.once('render', onRender);
-            }
-        });
-
-    } else if (operation[0] === 'wait') {
-        map.once(operation[1], function() {
-            applyOperations(map, operations.slice(1), callback);
-        });
-
-    } else {
-        try {
+    try {
+        var operation = operations && operations[0];
+        if (!operations || operations.length === 0) {
+            callback();
+
+        } else if (operation[0] === 'wait') {
+            var wait = function() {
+                if (map.loaded()) {
+                    applyOperations(map, operations.slice(1), callback);
+                } else {
+                    map.once('render', wait);
+                }
+            };
+            wait();
+
+        } else {
             map[operation[0]].apply(map, operation.slice(1));
-        } catch (e) {
-            console.error(e.stack);
-            throw e;
+            applyOperations(map, operations.slice(1), callback);
         }
-        applyOperations(map, operations.slice(1), callback);
+    } catch (e) {
```

Comment:
Good point. I will restructure to move `callback` out of the error handler.

> What happens if we don't do any exception handling here?

If we don't do any error handling and there is a synchronous error (there were several in my testing) some global exception handler (not the default one) gets triggered. This handler is awful for debugging because it prints the error message (i.e. "undefined is not a function") without any stracktrace.